### PR TITLE
Add all svg and refactor all image names, also improve some

### PR DESCRIPTION
**Pull request by dpizetta**
- [X] Update all svg files and generate new images;
- [X] Update all paths and names;
- [x] Review todos - order reversed;
- [X] Fix tree view branches;
- [X] Improve window actions icons, close, minimize, undock;
- [X] Enhance dock window actions;
- [X] Fix tab title tab borders, close icon;
- [X] Revise all widgets and paths;
- [X] Check scripts that generate generic images;
- [x] Add image in the contribute guide + more guidelines;
- [x] Create a checker for matching names and quantities of png, svg and links in the files - almost done.

### CODE REVIEW

**Comment by dpizetta** [comment]
@goanpeca could you take a look in the result? I'm in progress. Tks

**Comment by goanpeca** [comment]
Awesome work @dpizetta ! So far looks good. Ping me when finished so I can take a another look!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -223,14 +238,27 @@ For more information about those scripts, see their documentation.
...
+As a minimal guide to create new icons (svg) images, we list two main sources.
+
+ - [Material](https://material.io/design/iconography/product-icons.html#grid-keyline-shapes)
+ - [KDE](https://hig.kde.org/style/icon.html)
+
+Main characteristics are of SVG images are:
+
+- Size: 32px X 32px;
+- Border: 2px border space;
+- Line: 2px minimum thickness. Complementary 4px and 8px thickness;
```
Comment (goanpeca) [inline CONTRIBUTING.md]: I would add also that these lines and shape should align to the grid as much as possible (depending on the shape of course)

Comment (goanpeca) [reply CONTRIBUTING.md]: Maybe adding an image?

https://hig.kde.org/style/icon.html
Screen Shot 2019-05-24 at 17 10 50

**Comment by goanpeca** [comment]
We need to add a blacklist of svg, that should not be converted. `palette.svg` is one of them

**Comment by goanpeca** [comment]
The arrows on the `File Explorer` of spyder look too big. Maybe  those files need a different size (not squared?)

Screen Shot 2019-05-24 at 20 06 16

**Comment by goanpeca** [comment]
The close (used for tabs in spyder) looks small and not very readable

Screen Shot 2019-05-24 at 20 47 41

File: CONTRIBUTING.md (hunk 1 above)
Comment (dpizetta) [reply CONTRIBUTING.md]: Oh, yes, excellent! I will add.

**Comment by dpizetta** [comment]
> We need to add a blacklist of svg, that should not be converted. palette.svg is one of them

Yep! I'll provide! tks

> The arrows on the File Explorer of spyder look too big. Maybe those files need a different size (not squared?)

Actually we may don't need to keep the size changes in the file itself, but I need to change in the scss file. I have also changed many ones, but I unnoticed this one. Also, I have added the lines to the treeview that was there but not used before. However, it seems too full, may thinner lines. I noticed that people aren't using the lines now-a-days. I will print both examples so we can decide.

> The close (used for tabs in spyder) looks small and not very readable.

Could you check if there is any special piece of code that change the style in the spyder application? As we had the different sizes, people could be used some code adjust in style, that now is not needed. For example, the close-tab in the 'example' seems to be in a nice size.

>It should probably be a filled icon, and have the x be the transparent part (like circular ones were). 

I'm using the same for the close window and dock widget. If we remove the box around the X and make it thicker. Something like this one. 

image

I'm not a fan of keep the circle, because it leaves the interface heavier, but this is my opinion. 
@goanpeca  thanks for all the comments.

**Comment by dpizetta** [comment]
@goanpeca finished! Could you take another look? Tks

**Comment by goanpeca** [comment]
Yes! Will do in the afternoon

**Pull request merged**